tests/test_reed_solomon.py

```diff
@@ -0,0 +1,124 @@
+import pytest
```

Comment:
No tests for IncrementalDecoder? You can take it in a separate PR if you'd like since this is growing large and we should merge it before this blows up.

Reply:
Good catch on missing tests, we shouldn't add new content without minimal test coverage. I can't think of an easy way to recommend breaking this into multiple PRs, do you have a suggestion?

Reply:
I would actually suggest @rahulgovind to take the test up immediately after this PR and not add any more code with this PR. There is a lot of new code in this and we shouldn't add more..

Reply:
OK so let's merge this after the cosmetic fixes (no unnecessary file/line changes etc.), even without the tests for `IncrementalDecoder`, with an issue to add such tests in a subsequent PR